Update an existing SageMaker Endpoint with a newly trained Scikit-Learn Random Forest model

eitansela opened the pull request:
*Issue #, if available:*

*Description of changes:*
Added a new notebooks that shows how to update an existing SageMaker Endpoint with a newly trained Scikit-Learn Random Forest model using boto3.

*Testing done:*
Tested notebook end to end on SageMaker Notebooks and SageMaker Studio.

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

aaronmarkham reviewed (changes requested):
Can you update the "here" links:
1. have a period at the end of sentences that end w/ a link.
2. change "here" to something accessible and descriptive. Like:
```
Visit the scikit website for [more info on Scikit-Learn](https://scikit-learn.org/stable/index.html).
```

If this notebook is focused on boto3, perhaps the title could have that?
"Use boto3 to Update an Existing SageMaker Endpoint with a Newly Trained Scikit-Learn Random Forest Model"

Also, it might be worth pointing out that you're using some of the SageMaker Python SDK, but mostly focusing on boto3 as your preferred SDK to interface with the SageMaker service.

If the notebook's focus us "updating an endpoint", then why do the training here at all? Why not just load a tar.gz of the model we have on s3? If you want to show training with boto3, then let's update the title.

/advanced_functionality/ is a bit of a dumping ground and we'd like to clean this up. I suggest adding a top-level folder for boto3, then nesting this in an appropriate topic like /boto3/built-in-frameworks/scikitlearn/.

eitansela commented:
Hello @aaronmarkham , I modified the notebook by your comments:

- Renamed title to "Use boto3 to train and update an existing SageMaker Endpoint with a newly trained Scikit-Learn Random Forest Model".
- Added note about using some of the SageMaker Python SDK, but mostly focusing on boto3.
- Added note about Lambda, Airflow or Jenkins.
- Added a period at the end of sentences that end w/ a link.
- Changed "here" to something accessible and descriptive.
- Moved the notebook to the new boto3 folder location.
